modules/combined/test/tests/surface_tension_KKS/surface_tension_KKS.i

```diff
@@ -0,0 +1,412 @@
+    type = ElasticEnergyMaterial
+    base_name = matrix
+    f_name = fe_m
+    args = ' '
+  [../]
+# Total free energy of the matrix
+  [./Total_energy_matrix]
+    type = DerivativeSumMaterial
+    f_name = f_total_matrix
+    sum_materials = 'fm fe_m'
+    args = 'cm'
+  [../]
+
+  # Free energy of the precipitate phase
+  [./fp]
+    type = DerivativeParsedMaterial
+    f_name = fp
+    args = 'cp'
+    function = '6.55*(cp-0.235)^2'
+  [../]
+
+# Elastic energy of the precipitate
+  [./elastic_free_energy_p]
+    type = ElasticEnergyMaterial
+    base_name = ppt
+    f_name = fe_p
+    args = ' '
+  [../]
+
+# Total free energy of the precipitate
+  [./Total_energy_ppt]
+    type = DerivativeSumMaterial
+    f_name = f_total_ppt
+    sum_materials = 'fp fe_p'
+    args = 'cp'
+  [../]
+
+# Total elastic energy
+  [./Total_elastic_energy]
+    type = DerivativeTwoPhaseMaterial
+    eta = eta
+    f_name = f_el_mat
+    fa_name = fe_m
+    fb_name = fe_p
+    outputs = exodus
+    W = 0
+  [../]
+
+  # h(eta)
+  [./h_eta]
+    type = SwitchingFunctionMaterial
+    h_order = HIGH
+    eta = eta
+  [../]
+
+  # g(eta)
+  [./g_eta]
+    type = BarrierFunctionMaterial
+    g_order = SIMPLE
+    eta = eta
+    outputs = exodus
+  [../]
+
+  # constant properties
+  [./constants]
+    type = GenericConstantMaterial
+    prop_names  = 'M   L   kappa'
+    prop_values = '0.7 0.7 0.1365'
+  [../]
+
+  #Mechanical properties
+  [./Stiffness_matrix]
+    type = ComputeElasticityTensor
+    C_ijkl = '74.25 14.525'
+    base_name = matrix
+    fill_method = symmetric_isotropic
+  [../]
+  [./Stiffness_ppt]
+    type = ComputeElasticityTensor
+    C_ijkl = '74.25 14.525'
+    base_name = ppt
+    fill_method = symmetric_isotropic
+  [../]
+  [./strain_matrix]
+    type = ComputeRSphericalSmallStrain
+    base_name = matrix
+  [../]
+  [./strain_ppt]
+    type = ComputeRSphericalSmallStrain
+    base_name = ppt
+  [../]
+  [./stress_matrix]
+    type = ComputeLinearElasticStress
+    base_name = matrix
+  [../]
+  [./stress_ppt]
+    type = ComputeLinearElasticStress
+    base_name = ppt
+  [../]
+  [./global_stress]
+    type = TwoPhaseStressMaterial
+    base_A = matrix
+    base_B = ppt
+  [../]
+  [./interface_stress]
+    type = ComputeSurfaceTensionKKS
+    v = eta
+    kappa_name = kappa
+    w = 0.0033
+  [../]
+[]
+
+[BCs]
+  [./left_r]
+    type = PresetBC
+    variable = disp_x
+    boundary = left
+    value = 0
+  [../]
+[]
+
+#
+# Precondition using handcoded off-diagonal terms
+#
+[Preconditioning]
+  [./full]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = 'PJFNK'
+  petsc_options_iname = '-pc_type -sub_pc_type   -sub_pc_factor_shift_type'
+  petsc_options_value = 'asm       lu            nonzero'
+
+  l_max_its = 30
+  nl_max_its = 10
+  l_tol = 1.0e-4
+  nl_rel_tol = 1.0e-8
+  nl_abs_tol = 1.0e-9
+
+  num_steps = 2
+
+  [./TimeStepper]
+    type = IterationAdaptiveDT
```

Comment:
I suggest using a fixed dt in case future updates change the iteration count and this the time step, causing a diff